ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -2855,7 +2859,7 @@ and perform_exn ?subtask ?result (op : operation) (t : Xenops_task.task_handle)
               warn "VM %s has unexpectedly suspended" id ;
               [Vm.Shutdown]
           | Some Needs_softreset ->
-              B.VM.soft_reset t vm ; []
+              [Vm.Softreboot]
```

Comment:
I think you're missing the configurability, should this be `vm.Vm.on_softreboot`?